modules/evs/fsw/src/cfe_evs_utils.c

```diff
@@ -288,6 +289,105 @@ bool EVS_IsFiltered(EVS_AppData_t *AppDataPtr, uint16 EventID, uint16 EventType)
     return (Filtered);
 }
 
+/*----------------------------------------------------------------
+ *
+ * Function: EVS_CheckAndIncrementSquelchTokens
+ *
+ * Application-scope internal function
+ * See description in header file for argument/return detail
+ *
+ *-----------------------------------------------------------------*/
+bool EVS_CheckAndIncrementSquelchTokens(EVS_AppData_t *AppDataPtr)
+{
+    bool      NotSquelched = true;
+    OS_time_t CurrentTime;
+    int64     DeltaTimeMs;
+    int64     CreditCount;
+    char      AppName[OS_MAX_API_NAME];
+
+    /* Set maximum token credits to burst size */
+    const int32 UPPER_THRESHOLD = CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST;
+    /*
+     * Set lower threshold to stop decrementing
+     * Make this -CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST to add some hysteresis
+     * Events will resume (CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST /
+     * CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC + 1 /
+     * CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC) seconds after flooding stops if
+     * saturated
+     */
+    const int32 LOWER_THRESHOLD = -CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST;
+
+    if (CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST != 0)
+    {
+        /*
+         * We use a timer here since configurations are not guaranteed to send EVS HK wakeups at 1Hz
+         * Use a non-settable timer to prevent this from breaking w/ time changes
+         */
+        CFE_PSP_GetTime(&CurrentTime);
+        DeltaTimeMs = OS_TimeGetTotalMilliseconds(OS_TimeSubtract(CurrentTime, AppDataPtr->LastSquelchCreditableTime));
+
+        /* Calculate how many tokens to credit in elapsed time since last creditable event */
+        CreditCount = DeltaTimeMs * CFE_PLATFORM_EVS_APP_EVENTS_PER_SEC / 1000;
+
+        if (CreditCount > 0)
+        {
+            /* Update last squelch returned time if we credited any tokens*/
+            AppDataPtr->LastSquelchCreditableTime = CurrentTime;
+
+            /*
+             * Add Credits, to a maximum of UPPER_THRESHOLD
+             * Shouldn't rollover, as calculations are done in int64 space due to
+             * promotion rules then bounded before demotion
+             */
+            if (AppDataPtr->SquelchTokens > UPPER_THRESHOLD)
```

Comment:
Instead of using CFE_PLATFORM_EVS_MAX_APP_EVENT_BURST directly, set a global variable to this. Solves multiple problems for testability.